interfaces/builtin/bluez.go

```diff
@@ -0,0 +1,234 @@
+# Usage: reserved
+accept
+accept4
+bind
+connect
+getpeername
+getsockname
+getsockopt
+listen
+recv
+recvfrom
+recvmmsg
+recvmsg
+send
+sendmmsg
+sendmsg
+sendto
+setsockopt
+shutdown
+socketpair
+socket
+`)
+
+var bluezPermanentPlugSecComp = []byte(`
+# Description: Allow using bluez service. Reserved because this gives
+#  privileged access to the bluez service.
+# Usage: reserved
+
+# Can communicate with DBus system service
+connect
+getsockname
+recv
+recvmsg
+send
+sendto
+sendmsg
+socket
+`)
+
+var bluezPermanentPlugDBus = []byte(`
+<policy user="root">
+    <allow own="org.bluez"/>
+    <allow own="org.bluez.obex"/>
+    <allow send_destination="org.bluez"/>
+    <allow send_destination="org.bluez.obex"/>
+    <allow send_interface="org.bluez.Agent1"/>
+    <allow send_interface="org.bluez.ThermometerWatcher1"/>
+    <allow send_interface="org.bluez.AlertAgent1"/>
+    <allow send_interface="org.bluez.Profile1"/>
+    <allow send_interface="org.bluez.HeartRateWatcher1"/>
+    <allow send_interface="org.bluez.CyclingSpeedWatcher1"/>
+    <allow send_interface="org.bluez.GattCharacteristic1"/>
+    <allow send_interface="org.bluez.GattDescriptor1"/>
+    <allow send_interface="org.freedesktop.DBus.ObjectManager"/>
+    <allow send_interface="org.freedesktop.DBus.Properties"/>
+</policy>
+<policy context="default">
+    <deny send_destination="org.bluez"/>
+</policy>
+`)
+
+type BluezInterface struct{}
+
+func (iface *BluezInterface) Name() string {
+	return "bluez"
+}
+
+func (iface *BluezInterface) PermanentPlugSnippet(plug *interfaces.Plug, securitySystem interfaces.SecuritySystem) ([]byte, error) {
+	switch securitySystem {
+	case interfaces.SecurityDBus:
+		return bluezPermanentPlugDBus, nil
+	case interfaces.SecurityAppArmor:
+		return bluezPermanentPlugAppArmor, nil
+	case interfaces.SecuritySecComp:
+		return bluezPermanentPlugSecComp, nil
+	case interfaces.SecurityUDev:
+		return nil, nil
+	default:
+		return nil, interfaces.ErrUnknownSecurity
+	}
+}
+
+func (iface *BluezInterface) ConnectedPlugSnippet(plug *interfaces.Plug, slot *interfaces.Slot, securitySystem interfaces.SecuritySystem) ([]byte, error) {
+	return nil, nil
+}
+
+func (iface *BluezInterface) PermanentSlotSnippet(slot *interfaces.Slot, securitySystem interfaces.SecuritySystem) ([]byte, error) {
+	switch securitySystem {
+	case interfaces.SecurityAppArmor:
+		return bluezPermanentSlotAppArmor, nil
+	case interfaces.SecuritySecComp:
+		return bluezPermanentSlotSecComp, nil
+	case interfaces.SecurityUDev, interfaces.SecurityDBus:
+		return nil, nil
+	default:
+		return nil, interfaces.ErrUnknownSecurity
+	}
+}
+
+func (iface *BluezInterface) ConnectedSlotSnippet(plug *interfaces.Plug, slot *interfaces.Slot, securitySystem interfaces.SecuritySystem) ([]byte, error) {
+	return nil, nil
+}
+
+func (iface *BluezInterface) SanitizePlug(slot *interfaces.Plug) error {
+	return nil
+}
+
+func (iface *BluezInterface) SanitizeSlot(slot *interfaces.Slot) error {
+	return nil
```

Comment:
Once we have snap IDs I'd like this interface to be only available to the bluez snap. Perhaps that will be done with assertions, perhaps here. Not sure yet.